### PR TITLE
Add support of NC-Token to Nextcloud widget

### LINKED ISSUE #1122: [Feature Request] Add support for Nextcloud's NC-Token
### Description

Would be great to not be forced to use user+password for Nextcloud widget, but go with NC-Token instead:

image

### Other

_No response_

### PR DESCRIPTION
**Pull request by shalak**
## Proposed change

Adds support for Nextcloud's NC-Token authentication. This is safer than user+pass, as the only purpose of this token is to access the serverinfo endpoint, AFAIK.
image

Closes #1122

## Type of change

- [ ] New service widget
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (please explain)

## Checklist:

- [X] If adding a service widget or a change that requires it, I have added a corresponding PR to the [documentation](https://github.com/benphelps/homepage-docs) here: 
https://github.com/benphelps/homepage-docs/pull/60
- [x] If applicable, I have checked that all tests pass with e.g. `pnpm lint`.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.

### CODE REVIEW

File: src/utils/proxy/handlers/credentialed.js [hunk 1]
```diff
@@ -48,6 +48,14 @@ export default async function credentialedProxyHandler(req, res, map) {
         headers.Authorization = `Token ${widget.key}`;
       } else if (widget.type === "miniflux") {
         headers["X-Auth-Token"] = `${widget.key}`;
+      } else if (widget.type === "nextcloud") {
```
Comment (shalak) [inline src/utils/proxy/handlers/credentialed.js:51]: It would be a good idea to refactor the whole approach, so we could configure the `credentialedProxyHandler` behavior from `widget.js` files, that way it would keep more of a SOLID approach. However this is outside of the scope of this PR

**Comment by shamoon** [comment]
Cheers
